x/leverage/keeper/inspector.go

```diff
@@ -0,0 +1,363 @@
+	}
+	switch strings.ToLower(req.Sort) {
+	case "borrowed":
+		sorting = moreBorrowed(specific)
+		if sortMin.IsPositive() {
+			filters = append(filters, withMinBorrowedValue(sortMin, specific))
+		}
+	case "collateral":
+		sorting = moreCollateral(specific)
+		if sortMin.IsPositive() {
+			filters = append(filters, withMinCollateralValue(sortMin, specific))
+		}
+	case "danger":
+		sorting = moreDanger()
+		if sortMin.IsPositive() {
+			filters = append(filters, withMinDanger(sortMin))
+		}
+	case "ltv":
+		sorting = moreLTV()
+		if sortMin.IsPositive() {
+			filters = append(filters, withMinLTV(sortMin))
+		}
+	default:
+		// if no sort mode is specified, return all borrowers sorted by total collateral value
+		sorting = moreCollateral(false)
+	}
+
+	borrowers := k.filteredSortedBorrowers(ctx,
+		filters,
+		sorting,
+		req.Symbol,
+	)
+	return &types.QueryInspectResponse{Borrowers: borrowers}, nil
+}
+
+// Separated from grpc_query.go
+func (q Querier) InspectNeat(
+	goCtx context.Context,
+	req *types.QueryInspectNeat,
+) (*types.QueryInspectNeatResponse, error) {
+	if req == nil {
+		return nil, status.Error(codes.InvalidArgument, "empty request")
+	}
+
+	// The "all" symbol denom is converted to empty symbol denom
+	if strings.EqualFold(req.Symbol, "all") {
+		req.Symbol = ""
+	}
+
+	req2 := types.QueryInspect{
+		Symbol:  req.Symbol,
+		Mode:    req.Mode,
+		Sort:    req.Sort,
+		ModeMin: req.ModeMin,
+		SortMin: req.SortMin,
+	}
+
+	resp2, err := q.Inspect(goCtx, &req2)
+	if err != nil {
+		return nil, err
+	}
+
+	borrowers := []types.BorrowerSummaryNeat{}
+	for _, b := range resp2.Borrowers {
+		if b.SuppliedValue.IsPositive() {
+			borrowed := b.BorrowedValue
+			if req.Symbol != "" {
+				borrowed = b.SpecificBorrowValue
+			}
+			neat := types.BorrowerSummaryNeat{
+				Account:  b.Address,
+				Borrowed: neat(borrowed),
+				LTB:      neat(b.BorrowedValue.Quo(b.BorrowLimit)),
+				LTL:      neat(b.BorrowedValue.Quo(b.LiquidationThreshold)),
+				LTV:      neat(b.BorrowedValue.Quo(b.CollateralValue)),
+			}
+			borrowers = append(borrowers, neat)
+		}
+	}
+
+	return &types.QueryInspectNeatResponse{Borrowers: borrowers}, nil
+}
+
+// Separated from grpc_query.go
+func (q Querier) RiskData(
+	goCtx context.Context,
+	req *types.QueryRiskData,
+) (*types.QueryRiskDataResponse, error) {
+	if req == nil {
+		return nil, status.Error(codes.InvalidArgument, "empty request")
+	}
+
+	// This query is also disabled by default as a safety measure. Enable with liquidator queries.
+	if !q.Keeper.liquidatorQueryEnabled {
+		return nil, types.ErrNotLiquidatorNode
+	}
+	k, ctx := q.Keeper, sdk.UnwrapSDKContext(goCtx)
+	borrowers := k.riskData(ctx)
+
+	return &types.QueryRiskDataResponse{Borrowers: borrowers}, nil
+}
+
+// neat truncates an sdk.Dec to a common-sense precision based on its size and converts it to float.
+// This makes a big difference in readability when using the CLI.
+func neat(num sdk.Dec) float64 {
```

Comment:
could you add few tests to this function?

Reply:
added